Controller/Adminhtml/Menu/Edit.php

```diff
@@ -1,58 +1,100 @@
 <?php
 
+declare(strict_types=1);
+
 namespace Snowdog\Menu\Controller\Adminhtml\Menu;
 
-use Magento\Backend\App\Action;
-use Magento\Framework\App\ResponseInterface;
-use Magento\Framework\Controller\ResultFactory;
-use Magento\Framework\Exception\NoSuchEntityException;
+use Snowdog\Menu\Controller\Adminhtml\AbstractMenu;
+use Magento\Framework\App\Action\HttpGetActionInterface;
+use Magento\Backend\App\Action\Context;
 use Magento\Framework\Registry;
+use Magento\Framework\View\Result\PageFactory;
 use Snowdog\Menu\Api\MenuRepositoryInterface;
+use Snowdog\Menu\Model\MenuFactory;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\Controller\ResultInterface;
+use Magento\Backend\Model\View\Result\Redirect;
+use Magento\Backend\Model\View\Result\Page;
 
-class Edit extends Action
+class Edit extends AbstractMenu implements HttpGetActionInterface
 {
     const REGISTRY_CODE = 'snowmenu_menu';
+
     /**
      * @var MenuRepositoryInterface
      */
     private $menuRepository;
+
+    /**
+     * @var PageFactory
+     */
+    private $resultPageFactory;
+
+    /**
+     * @var MenuFactory
+     */
+    private $menuFactory;
+
     /**
      * @var Registry
      */
-    private $registry;
+    private $coreRegistry;
 
-    public function __construct(Action\Context $context, MenuRepositoryInterface $menuRepository, Registry $registry)
-    {
-        parent::__construct($context);
+    /**
+     * @param Context $context
+     * @param Registry $coreRegistry
+     * @param PageFactory $resultPageFactory
+     * @param MenuRepositoryInterface $menuRepository
+     * @param MenuFactory $menuFactory
+     */
+    public function __construct(
+        Context $context,
+        Registry $coreRegistry,
+        PageFactory $resultPageFactory,
+        MenuRepositoryInterface $menuRepository,
+        MenuFactory $menuFactory
+    ) {
+        $this->resultPageFactory = $resultPageFactory;
         $this->menuRepository = $menuRepository;
-        $this->registry = $registry;
+        $this->menuFactory = $menuFactory;
+        $this->coreRegistry = $coreRegistry;
+        parent::__construct(
+            $context
+        );
     }
 
     /**
-     * Dispatch request
-     *
-     * @return \Magento\Framework\Controller\ResultInterface|ResponseInterface
-     * @throws \Magento\Framework\Exception\NotFoundException
+     * @SuppressWarnings(PHPMD.NPathComplexity)
```

Comment:
I think you should refactor the method to avoid this statement

Reply:
Removed it. Seems we don't need it.